llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -1239,6 +1240,20 @@ AliasResult BasicAAResult::aliasGEP(
     }
   }
 
+  // If the difference between pointers is Offset +<nuw> Indices then we know
+  // that the addition does not wrap the pointer index type (add nuw) and the
+  // constant Offset is a lower bound on the distance between the pointers. We
+  // can then prove NoAlias via Offset u>= VLeftSize.
+  //    +                +                     +
+  //    | BaseOffset     |   +<nuw> Indices    |
+  //    ---------------->|-------------------->|
+  //    |-->VLeftSize    |                     |-------> VRightSize
```

Comment:
```suggestion
  //    |-->V2Size    |                     |-------> V1Size
```
I think? WOuld be good to have the names here line up with the variable names.

Reply:
Done.